client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -935,6 +935,33 @@ Undocumented.prototype.setPrimaryDomain = function( siteId, domain, fn ) {
 	this.wpcom.req.post( '/sites/' + siteId + '/domains/primary', {}, { domain: domain }, fn );
 };
 
+/**
+ * Fetch preview markup for a site
+ *
+ * @param {int} siteId The site ID
+ * @param {string} slug Optional. The theme slug to preview
+ * @param {object} postData Optional. The customization data to send
+ * @return {Promise} A Promise to resolve when complete
+ */
+Undocumented.prototype.fetchPreviewMarkup = function( siteId, slug, postData ) {
+	debug( '/sites/:site_id/previews/mine' );
+	return new Promise( ( resolve, reject ) => {
+		const endpoint = `/sites/${siteId}/previews/mine`;
+		const query = { path: slug };
+		const isPreviewCustomized = ( postData && Object.keys( postData ).length > 0 );
+		const { post, get } = this.wpcom.req;
```

Comment:
It still makes me uncomfortable, though. :) And, actually, I'm not convinced that there's any magic binding happening: see [Babel output](https://babeljs.io/repl/#?evaluate=true&presets=es2015%2Creact%2Cstage-2&experimental=true&loose=false&spec=false&code=function%20bar()%20%7B%0A%20%20const%20%7B%20post%2C%20get%20%7D%20%3D%20this.wpcom.req%3B%0A%20%20post(%20foo%20)%3B%0A%7D).

I'd keep the method calls attached to the whole context tree (`this.wpcom.req.post( bla )`). Then again, I am not bothered by three-line ternaries. ;)